### DISCUSSION TITLE
Manim CE Question — ApplyFuncExample

### DISCUSSION THREAD

**Discussion by kinggraham336** [3b1b/manim#1708] [Q&A] [answered]
### Please Explain

_I am currently learning the **Manim Community v.0.14.0** library._ I am a beginner trying to read and write this code so as to understand it. 

**Animation link**: https://user-images.githubusercontent.com/90053391/149587727-e51cf423-f871-4770-b397-807d1363750b.mp4
**Link to Manim CE Documentation:** https://docs.manim.community/en/stable/reference/manim.mobject.mobject.Mobject.html?highlight=copy()#manim.mobject.mobject.Mobject.copy

https://user-images.githubusercontent.com/90053391/149588843-5bad0379-4b54-4ca9-bb7e-26a89da753d1.mp4

This is the code I was analyzing, copied directly from example in the **Animation Link** above:

                              `from manim import *
                              
                              class ApplyFuncExample(Scene):
                                  def construct(self):
                                      circ = Circle().scale(1.5)
                                      circ_ref = circ.copy()
                                      **circ.apply_complex_function(
                                          lambda x: np.exp(x*1j)
                                      )
                                      t = ValueTracker(0)**
                                      circ.add_updater(
                                          lambda x: x.become(circ_ref.copy().apply_complex_function(
                                              lambda x: np.exp(x+t.get_value()*1j)
                                          )).set_color(BLUE)
                                      )
                                      self.add(circ_ref)
                                      self.play(TransformFromCopy(circ_ref, circ))
                                      self.play(t.animate.set_value(TAU), run_time=3)`

The area in the code which I am asking about is in bold. My question is:

What is this lambda function creating? And how does it express itself in the red circle? I see no point for it being there, and I don't understand where it shows up on the animation. **If someone can explain to me how this particular code snippet is structured, what it is doing, and how it appears on the animation, I would appreciate it.**

**Comment by TonyCrane** [comment, marked as answer]
you are using CE version, so you are supposed to ask this question on their discord channel or open an issue in their repo instead of here.
